src/args.rs

```diff
@@ -0,0 +1,441 @@
+#![warn(rust_2018_idioms)]
+
+use std::{fmt, iter, num::ParseIntError, ops::RangeInclusive, path::PathBuf, str::FromStr};
+
+use clap::{Command, Parser, error::ErrorKind};
+use core_affinity::CoreId;
+use either::Either;
+use serde::{
+	de::{Deserialize, SeqAccess, Visitor},
+	*,
```

Comment:
technically, this brings into scope:
```rust
use serde::{
    de::{SeqAccess, Visitor},
    ser,
    forward_to_deserialize_any,
    Deserialize,
    Deserializer,
    Serialize,
    Serializer,
}
```

I think it would make sense to list the required items explicitly.

Reply:
Yeah, this is a very early WIP version and I'm not planning to keep it like that, this is an early concept and I'm just moving things around.